src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoCacheService.cs

```diff
@@ -26,7 +26,7 @@ namespace Microsoft.CodeAnalysis.FindSymbols.SymbolTree
     internal sealed partial class SymbolTreeInfoCacheService : IWorkspaceService
     {
         private readonly ConcurrentDictionary<ProjectId, SymbolTreeInfo> _projectIdToInfo = new();
-        private readonly ConcurrentDictionary<MetadataId, MetadataInfo> _metadataIdToInfo = new();
```

Comment:
this was not safe, different PortableExecutableReferences can have hte same MetadataId.

Reply:
this could lead to removing the mapping once one of several PERefs was removed that had the same ID.